### PR TITLE
[ADF-4499] Form definition selector should only display standAlone co…

### PR DESCRIPTION
**Pull request by davidcanonieto**
…mpatible forms

**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://issues.alfresco.com/jira/browse/ADF-4699
Form definition selector displays all forms contained in a certain app

**What is the new behaviour?**
The form definition selector should filter by forms that are standAlone task compatible.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
https://issues.alfresco.com/jira/browse/ADF-4699

### CODE REVIEW

File: lib/process-services-cloud/src/lib/form/services/form-definition-selector-cloud.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        oauth2Auth.callCustomApi.and.returnValue(Promise.resolve(responseBody));
+
+        service.getForms(appName).subscribe((result) => {
+            expect(result).toBeDefined();
+            expect(result.length).toBe(3);
+            done();
+        });
+    });
+
+    it('hould fetch only standAlone enabled forms when getStandAloneTaskForms is called', (done) => {
```
Comment (gmandakini) [inline lib/process-services-cloud/src/lib/form/services/form-definition-selector-cloud.service.spec.ts]: hould => Should

File: lib/process-services-cloud/src/lib/form/services/form-definition-selector-cloud.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            version: 0,
+            standAlone: false
+        }
+    },
+    {
+        formRepresentation: {
+            id: 'form-de8895be-d0d7-4434-beef-53453453452',
+            name: 'Form 3',
+            description: '',
+            version: 0
```
Comment (gmandakini) [inline lib/process-services-cloud/src/lib/form/services/form-definition-selector-cloud.service.spec.ts]: is there a scenario possible in reality that there is a response json coming back with standAlone variable but no value assigned to it? or does it always have a default value? 
if so then what is the default flag..

Comment (eromano) [reply lib/process-services-cloud/src/lib/form/services/form-definition-selector-cloud.service.spec.ts]: the check in the filter on undefined added by David should save this scenario

**Comment by eromano** [comment]
| am merging the PR also if there are two unstable test failing. We need to create a task to make them stable

**Pull request merged**